Sources/App/Features/Presentations/Controllers/PresentationAPIController.swift

```diff
@@ -0,0 +1,75 @@
+import Vapor
+import Fluent
+
+struct PresentationAPIController: RouteCollection {
+    
+    private struct FormInput: Content {
+        let speaker: String
+        let event: String
+        let title: String
+        let synopsis: String
+    }
+    
+    func boot(routes: RoutesBuilder) throws {
+        routes.post("", use: onPost)
+        routes.post(":id", use: onEdit)
+    }
+
+    private func onPost(request: Request) async throws -> Response {
+        let input = try request.content.decode(FormInput.self)
+        
+        guard let speaker = try await Speaker.find(UUID(uuidString: input.speaker), on: request.db) else {
+            return request.redirect(to: "/admin")
+        }
+        
+        guard let event = try await Event.find(.init(uuidString: input.event), on: request.db) else {
+            return request.redirect(to: "/admin")
+        }
+        
+        guard let eventID = event.id else {
+            return request.redirect(to: "/admin")
+        }
+        
+        let presentation = Presentation(
+            id: .generateRandom(),
```

Comment:
Should a randomised ID not be default if you don't provide one?

Reply:
Not sure how this works, this is copied over from the initial `presentations` branch for creation, just looks like I added it I think cause of the file name / structure change. 

I'll take a look though and if so might be good to roll out across all model creations.

Reply:
Just thought - if the above is the case to move it under admin, might be worth a whole folder structure shuffle to put `Features/Presentations/...` under `Admin/`?